### PR TITLE
Correctly replace debug lexical scope of instruction

### PR DESCRIPTION
**Pull request by jaebaek**
When we update OpenCL.DebugInfo.100 lexical scopes e.g., DebugFunction,
we have to replace DebugScope of each instruction that uses the lexical
scope correctly.

### CODE REVIEW

File: source/opt/ir_context.h [hunk 1]
```diff
@@ -576,6 +576,24 @@ class IRContext {
...
+  // Updates DebugInlinedAt of |inst| and analyzes use information
+  // of DebugInlinedAt.
+  void UpdateDebugInlinedAt(Instruction* inst, uint32_t dbg_inlined_at) {
+    inst->UpdateDebugInlinedAt(dbg_inlined_at);
+    if (AreAnalysesValid(kAnalysisDebugInfo)) {
+      get_debug_info_mgr()->AnalyzeDebugInst(inst);
+    }
+  }
+
+  // Updates DebugScope and line of |inst| and analyzes use information
```
Comment (s-perron) [inline source/opt/ir_context.h:588]: You can update the analysis in `Instruction::UpdateDebugInfoFrom`.  The instruction class has access to the context.

File: source/opt/ir_context.h [hunk 2]
```diff
@@ -576,6 +576,24 @@ class IRContext {
     return GetFunction(inst->result_id());
   }
 
+  // Updates DebugInlinedAt of |inst| and analyzes use information
+  // of DebugInlinedAt.
+  void UpdateDebugInlinedAt(Instruction* inst, uint32_t dbg_inlined_at) {
```
Comment (s-perron) [inline source/opt/ir_context.h:581]: I don't think you need this function.  You can move the update of the debug info manager to the instruction classes version of the function.

Comment (jaebaek) [reply source/opt/ir_context.h]: done

File: source/opt/ir_context.h (hunk 1 above)
Comment (jaebaek) [reply source/opt/ir_context.h]: done

**Pull request merged**